runtime/lua/vscode/sync-options.lua

```diff
@@ -46,68 +65,67 @@ end
 ---@param buf number
 ---@param opts EditorOptions
 local function set_options(buf, opts)
-  -- Due to debounce of editor options change event, in some special cases, buffer
-  -- may be deleted when starting to set options.
   if not api.nvim_buf_is_valid(buf) then
     return
   end
+
   api.nvim_buf_set_var(buf, "vscode_editor_options", opts)
-  if vim.bo[buf].ts ~= opts.tabSize then
-    vim.bo[buf].ts = opts.tabSize
-    vim.bo[buf].sw = opts.tabSize
-  end
-  if vim.bo[buf].et ~= opts.insertSpaces then
-    vim.bo[buf].et = opts.insertSpaces
-  end
 
-  local win
-  for _, w in ipairs(api.nvim_list_wins()) do
-    local win_buf = api.nvim_win_get_buf(w)
-    if win_buf == buf then
-      win = w
-      break
-    end
-  end
+  M.buf_set_option(buf, "tabstop", opts.tabSize)
+  M.buf_set_option(buf, "shiftwidth", opts.tabSize)
+  M.buf_set_option(buf, "expandtab", opts.insertSpaces)
 
-  if win then
+  local win = api.nvim_get_current_win()
+  if api.nvim_win_get_buf(win) == buf then
     set_number(win, opts.lineNumbers)
   end
 end
 
 ---Check changes from nvim
 ---Set vscode options
-local function _check_options()
-  ---@type EditorOptions?
-  local opts = vim.b.vscode_editor_options
-  if not opts then -- should not happen
+local function check_options()
+  local vscode_opts = vim.b.vscode_editor_options
+  if not vscode_opts then -- should not happen
     return
   end
-  if not vim.b.vscode_editor_options_first_checked then -- load the defaults
+
+  if not vim.b.vscode_editor_options_first_checked then --load the defaults
     vim.b.vscode_editor_options_first_checked = true
-    vim.bo.ts = opts.tabSize
-    vim.bo.sw = opts.tabSize
-    vim.bo.expandtab = opts.insertSpaces
-    set_number(0, opts.lineNumbers)
+    M.buf_set_option(0, "tabstop", vscode_opts.tabSize)
+    M.buf_set_option(0, "shiftwidth", vscode_opts.tabSize)
+    M.buf_set_option(0, "expandtab", vscode_opts.insertSpaces)
+    set_number(0, vscode_opts.lineNumbers)
     return
   end
 
-  local ts, sw, et = vim.bo.ts, vim.bo.sw, vim.bo.et
-  local lineNumbers = get_number_style(vim.wo.nu, vim.wo.rnu)
+  local buf = api.nvim_get_current_buf()
+  local win = api.nvim_get_current_win()
+  local ts = api.nvim_get_option_value("tabstop", { buf = buf })
+  local sw = api.nvim_get_option_value("shiftwidth", { buf = buf })
+  local et = api.nvim_get_option_value("expandtab", { buf = buf })
```

Comment:
This part is still more concise with the original `vim.bo` and `vim.wo`. What do you think?

Reply:
Yeah, original is simpler.